### PR TITLE
Cleaning up jsonnet parser logic

### PR DESCRIPTION
**Pull request by IanWoodard**
Addressing review comments from #2. 
* Moving away from using `System.out.println` to log errors.
* Adding a custom exception for Jsonnet eval errors. 
* Refactoring `compileJsonnet` methods to reduce duplicate code.
* Adding JavaDoc to better document the code.

### CODE REVIEW

File: src/main/java/cd/go/plugin/config/yaml/JsonnetEvalException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package cd.go.plugin.config.yaml;
+
+public class JsonnetEvalException extends RuntimeException {
```
Comment (mattgauntseo-sentry) [inline src/main/java/cd/go/plugin/config/yaml/JsonnetEvalException.java:3]: Is there any value of having this exception class over using RuntimeException directly?

Comment (IanWoodard) [reply src/main/java/cd/go/plugin/config/yaml/JsonnetEvalException.java:3]: I feel that it provides a few clear benefits:

- Better program flow control (catching these exceptions specifically)
- Better readability (it is clear what they error refers to)

Comment (mattgauntseo-sentry) [reply src/main/java/cd/go/plugin/config/yaml/JsonnetEvalException.java:3]: Catching these exceptions specifically is a good point.

**Review by mattgauntseo-sentry — APPROVED** [review]

File: src/main/java/cd/go/plugin/config/yaml/JsonnetConfigParser.java [hunk 2]
```diff
@@ -22,55 +36,68 @@ public JsonConfigCollection parseFiles(File baseDir, String[] files) {
...
+     * @return InputStream containing the compiled jsonnet
+     * @throws JsonnetEvalException if jrsonnet exits with an error
+     */
+    private InputStream compileJsonnet(String... command) throws JsonnetEvalException {
+        String[] commandWithArgs = new String[command.length + 1];
+        commandWithArgs[0] = "jrsonnet";
+        System.arraycopy(command, 0, commandWithArgs, 1, command.length);
         try {
-            String inputString = new String(input.readAllBytes());
-            ProcessBuilder pb = new ProcessBuilder("jrsonnet", "--exec", inputString, "--format", "yaml");
```
Comment (bukzor-sentryio) [inline src/main/java/cd/go/plugin/config/yaml/JsonnetConfigParser.java:62]: redirectError([Redirect.INHERIT](https://docs.oracle.com/javase/8/docs/api/java/lang/ProcessBuilder.Redirect.html#INHERIT))

Comment (bukzor-sentryio) [reply src/main/java/cd/go/plugin/config/yaml/JsonnetConfigParser.java:62]: either un-redirect stderr or else tack on the stderr output to the exception

Comment (bukzor-sentryio) [reply src/main/java/cd/go/plugin/config/yaml/JsonnetConfigParser.java:62]: Someday soon you should add this test back where it was: shouldRespondSuccessWithErrorMessagesToParseDirectoryRequestWhenParsingErrorCaseFile

**Review by bukzor-sentryio — APPROVED** [review]

**Pull request merged**